_inc/lib/class.core-rest-api-endpoints.php

```diff
@@ -283,41 +281,17 @@ public static function register_endpoints() {
 		) );
 	}
 
-	public static function route( $path, $classname, $method,
-		$constructor_arguments = NULL,
-		$endpoint_arguments = NULL
-	) {
-		if ( ! empty( $constructor_arguments ) ) {
-			$endpoint = new $classname( $constructor_arguments );
-		} else {
-			$endpoint = new $classname();
-		}
-
-		$parameters = array(
-			'methods' => $method,
-			'callback' => array( $endpoint, 'process' ),
-			'permission_callback' => array( $endpoint, 'can_request' )
-		);
-
-		if ( ! empty( $endpoint_arguments ) ) {
-			$parameters['args'] = $endpoint_arguments;
-		}
-
-		register_rest_route( 'jetpack/v4', $path, $parameters );
-	}
-
 	/**
 	 * Handles dismissing of Jetpack Notices
 	 *
 	 * @since 4.3.0
 	 *
 	 * @return array|wp-error
 	 */
-	public static function dismiss_notice( $data ) {
-		$notice = $data['notice'];
-		$param = $data->get_json_params();
+	public static function dismiss_notice( $request ) {
```

Comment:
Since we're renaming `$data` to `$request` can we also fix the phpDoc and add the following everywhere where it applies?
```
@param WP_REST_Request $request The request sent to the WP REST API.
```